analyzer/src/main/kotlin/managers/NPM.kt

```diff
@@ -187,7 +187,12 @@ class NPM : PackageManager() {
 
             val declaredLicenses = sortedSetOf<String>()
             setOf(json["license"]).mapNotNullTo(declaredLicenses) {
-                it?.asText()
+                it?.let {
+                    if (it.has("type")) it["type"].asText() else it.asText()
```

Comment:
A bit convoluted 😉 Inspired by similar code in `parseVcsInfo()` (which also looks for an optional nested `type`), how about instead:

            val declaredLicenses = sortedSetOf<String>()

            json["licenses"]?.mapNotNullTo(declaredLicenses) { licenseNode ->
                licenseNode["type"]?.asText()
            }

            json["license"]?.let { license ->
                val type = license.textValue() ?: license["type"].asTextOrEmpty()
                declaredLicenses.add(type)
            }

Also note to look for "licenses" (plural) first to not overwrite single license findings added via `declaredLicenses.add(type)`.

Reply:
Done, but I kept the order because `mapNotNullTo` also calls `add`, so no risk of overwriting.